troi/parse_prompt.py

```diff
@@ -0,0 +1,113 @@
+#!/usr/bin/env python3
+
+import sys
+from uuid import UUID
+
+import pyparsing as pp
+import pyparsing.exceptions
+
+OPTIONS = ["easy", "hard", "medium", "and", "or", "nosim"]
+
+
+class ParseError(Exception):
+    pass
+
+
+def build_parser():
```

Comment:
I'll have to read pyparsing docs to understand how this works but if this works then sounds fine to me.